.github/workflows/kit-bootstrap.yml

```diff
@@ -0,0 +1,135 @@
+name: This workflow automates the build, testing, and bootstrapping of a test environment on Kubernetes using KIT CLI
+on:
+  workflow_run:
+    workflows: [trigger-kit-CI]
+    types: [completed]
+permissions:
+  id-token: write
+  pull-requests: write
+  contents: read
+jobs:
+  kit:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/install-go-and-dependencies
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::953421922360:role/kit-Github-workflow
+          role-session-name: githubworkflowsession
+          aws-region: us-west-2
+      - name: Install kubectl
+        run: |
+          if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+            curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -L -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl"
+          elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+            curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -L -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/darwin/amd64/kubectl"  
+          fi
+          chmod +x kubectl
+          sudo mv kubectl /usr/local/bin/
+      - name: Builds the KIT CLI ,Provisions the Kit Environment,KIT guest cluster and deletes.
+        run: |
+          cd ${HOME}/work/kubernetes-iteration-toolkit/kubernetes-iteration-toolkit/substrate/cmd/kitctl
+          go build  .
+          if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+            os_type="linux"    
+          elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+            os_type="darwin"
+          fi 
+          ./kitctl bootstrap kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}
+          export KUBECONFIG=${HOME}/.kit/env/kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}/etc/kubernetes/admin.conf
+          kubectl get pods -A
+          for i in 21 22 23 24 25; do
+            GUEST_CLUSTER_NAME="guest-${os_type}-1-${i}-${{ github.run_id }}-${{github.run_attempt}}"
+            K8S_VERSION=$(echo ${GUEST_CLUSTER_NAME} | sed 's/guest-linux-1-//;s/-.*//;s/^/1./')
```

Comment:
I don't understand what you are doing here. How can you get `K8S_VERSION` from `GUEST_CLUSTER_NAME `?

Reply:
Example 
GUEST_CLUSTER_NAME=guest-linux-1-25-2846287626-8
K8S_VERSION=$(echo ${GUEST_CLUSTER_NAME} | sed 's/guest-linux-1-//;s/-.*//;s/^/1./')
Here we want the k8 version only in  string type first it removes the prefix "guest-linux-1-" ,removes the suffix after the first dash "-" and adds the prefix "1." to the extracted version number

Reply:
Why make it so complicated when `${i}` holds the information for you.
```
K8S_VERSION=1-${i}
```